### ISSUE TITLE
Split jetpack_file_data based off of autoload needs

### ISSUE THREAD

**Issue by mqudsi** [Automattic/jetpack#8268]
On a rather small site I'm administering, the `option_value` field for `jetpack_file_data` option in `wp_options` is 61KB. The problem is that this option has `autoload` set to `yes`, meaning *each and every* pageload ends up transferring 61KB - most of which is wholly unneeded - from the database.

This option contains things like the description for various JetPack modules, the original and supported WP versions for different modules, and even options for settings that are not present in the current installation such as ad management.

No matter what, simply serializing an object containing all this data and dumping it into a single kv pair in the database is ridiculous and screams bad design in the first place, but having it such that this content is loaded for each and every page load is just too much.

May I advise that the serialized contents of this field be pared down to the bare minimum, that fields that are not actually required to be autoloaded are moved to a different kv, and that in general, the size of the serialized content is kept in mind wherever possible. The problem with using PHP's generic serialization is that it is fully-typed by default, field names of objects are serialized too - not just their values (a la protobuff and others).

**Comment by jeherve** [comment]
cc @nickdaugherty who worked on this part of the code not that long ago in #8056, for some other performance improvements.

**Comment by nickdaugherty** [comment]
Yep, the `jetpack_file_data` has been moved to transients, which should alleviate the autoloading concern. This also lets it be served by a caching drop-in to take this load off the DB.

**Comment by mqudsi** [comment]
I think old `jetpack_file_data` are not removed from the db then? Unless the move to transients wasn’t in place the time this issue was opened.

**Comment by nickdaugherty** [comment]
That's correct - the code in #8056 simply switched the calls from options to transients, but the old option could still exist (and is safe to delete).

**Comment by mqudsi** [comment]
Perhaps this is worth adding a cleanup script to the db upgrade code for the next release?

**Comment by krunkosaurus** [comment]
Hi guys, I just too noticed `jetpack_file_data` in `wd_options` is 60,000 chars with autoload=yes in a Wordpress community I manage. Googled around and found this ticket. 

 It's good to know that we can delete this as it gets loaded with every single request. Agree there should be some cleanup script for it.

**Comment by jb510** [comment]
+1 on a cleaning this up, just noticed it autoloading 60k on a tiny site...

**Comment by dcx15** [comment]
Same issue in the year 2020, this option is still lingering around in a site's database. Glad to hear I can delete it though.

**Comment by valeciawolf34** [comment]
Close

**Comment by valeciawolf34** [comment]
https://drive.google.com/file/d/1jBLGy7sfFu9ILJI2m8Wagrr3rXOa7ZYG/view?usp=drivesdk

Valecia G Rodriguez

**Comment by iandunn** [comment]
I'm sure there's a cost/benefit analysis to make here, and that's totally understandable. 

FWIW, though, one case that hasn't been mentioned yet are cron jobs that loop through all the sites in a Multisite network. If there are ~850 sites created before the switch, then that can consume _roughly_ 21mb of RAM just for this option.

**Comment by galbaras** [comment]
+1
